DB models setup + relationships

svmoll opened the pull request:
Wrote DB models in the models.models.py folder rather than for each entity. Can be altered if decided. Tables relationship written with relationship and backref. Comments:
**Contacts:** in DrawSQL we have user_username but to be an Integer. Currently I have set it up as user_username String in the model. 
**Cards**: design_path is currently nullable. 
**Transactions:** transacation_date is None by default until populated when transaction is completed, as per our latest logic. Some more comments for future reference included as well.

ina-ene reviewed (approved):
There are some details and corrections that will be addressed in a different issue. Here are some of them
-contacts is a junction table using username both for user and contact and not phone number; contacts model needs revision as a result of it being a junction table, primary_key=True need to be added for both columns 
-password in User has repeated nullable=False
-there are index=True in the tables which are not necessarily needed in some places
-Datetime.datetime needs a fix

svmoll commented:
- contacts - I was not clear it's a junction table, my bad. We will talk in the discussion.
- password - Agree.
- indexing - Agree. Removed from user.email, cards.card_number, cards.expiration_date as they will not be used for sorting/filtering for now, according to initial project guidelines. Added to categories.name, all primary and foreign keys to be consistent with guidelines.
- datetime - not sure what fix is suggested.

Keen to train restoring branch, committing the changes and redoing the process. Changes are currently only locally.

svmoll commented:
contacts - Clarified. 
indexing - We will have _index=True_ only on users.phone_number not to make the architecture heavier.
datetime - I had a mistake.

Fixes will be pushed by Kari, together with other code.
